### ISSUE TITLE
/etc/config is not updated after Fedora upgrade.

### ISSUE THREAD

**Issue by az-z** [rpm-software-management/mock#563]
### Short description of the problem

As per discussion on the buildysys:
Pavel :
I was corrected by @msuchy now, this is not %config(noreplace) because it
basically brings arch-specific file in noarch package.

So there's pretty complicated %post scriptlet for this, which doesn't ever
replace the old symlink, and only creates `rpmnew` file.

### Output of `rpm -q mock`
mock-2.1-1.fc31.noarch

### Steps to reproduce issue

1.upgrade from FC30 to FC31

### Any additional notes

<details>
 I just upgraded from FC30 to FC 31.
 it seems that mock's "default.cfg" is still pointing to the FC30 config.

cd /etc/mock/
[az@dell5000 mock]$ ls -l ./default.cfg
lrwxrwxrwx 1 root root 20 Nov 24 12:52 ./default.cfg -> fedora-30-x86_64.cfg

[az@dell5000 mock]$ uname -a
Linux dell5000 5.5.16-200.fc31.x86_64 #1 ...
...
[root@dell5000 ~]# dnf whatprovides /etc/mock/default.cfg
Last metadata expiration check: 1:41:15 ago on Sat 18 Apr 2020 02:52:52 PM EDT.
mock-core-configs-31.6-1.fc31.noarch : Mock core config files basic chroots
Repo        : fedora
Matched from:
Filename    : /etc/mock/default.cfg

mock-core-configs-32.4-1.fc31.noarch : Mock core config files basic chroots
Repo        : @System
Matched from:
Filename    : /etc/mock/default.cfg

mock-core-configs-32.4-1.fc31.noarch : Mock core config files basic chroots
Repo        : updates
Matched from:
Filename    : /etc/mock/default.cfg
</details>

**Comment by xsuchy** [comment]
The file `/etc/mock/default.cfg` is symlink to e.g. fedora-29-x86_64.cfg. The problem is that after upgrade to F31 we can hardly guess whether the symlink is the default one (and user upgrading from F29->F31) or whether it is custom one. E.g. user is upgrading from F30->F31 but they wants to have default config pointed to `fedora-29-x86_64.cfg`

Note that `default.cfg` is the config used if you run mock without `-r` option.

So my original intention was that Mock should not play a smarty and simply create .rpmnew (exactly what %config(noreplace) would does) and let user handle it - simplest ways is `rpmconf -o mock-core-configs`

**Comment by praiskup** [comment]
Good point we can not detect the previous version in new package %post.

But we could e.g. remove the default symlink in `%postun of old package`
and create the new one in `%posttrans of new package`.  Per [docs](https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#ordering).

I don't know.  What if we provided arch-specific subpackage with default
symlink?  Or what if mock-core-configs was arch-specific package actually?

**Comment by praiskup** [comment]
Also the default.cfg could be non-symlink file doing appropriate
include, if we had a good host detection.  Having such detection
in hand would e.g. allow us to remove `etc/hosts-overrides` hack.

**Comment by az-z** [comment]
The point is well taken. We can argue either way, for example,  we can say
that "default " should be default for the current env as in host is 31, so
defaults to 31. As an example,  if i have a custom kernel on 30 and then
upgrade to 31, my default kernel gets upgraded too..

**Comment by praiskup** [comment]
I'd propose removing the `default.cfg` from `mock-core-configs`.  If user installs the file manually, it is fine (user's responsibility).  Otherwise, if default.cfg is non-existent - we can detect the default chroot at mock runtime.

**Comment by xsuchy** [comment]
> I'd propose removing the default.cfg from mock-core-configs. 

-1 to this. I am used to this. And I guess many others.

> Also the default.cfg could be non-symlink file doing appropriate
include,

+1 to this.
we can make default.cfg simple `include($FILE)` where FILE can guessed during buildtime. It will become ordinary file. And can be maintained by rpmconf as any other file. (although rpmconf can deal with symlinks too).

**Comment by Conan-Kudo** [comment]
If we teach mock how to detect automatically, we should just remove `default.cfg` symlink. If someone needs it, we should document how they can set the default in their user-local mock configuration, similar to how they configure their default EL variant of the EPEL configs.
